### PR TITLE
xds: drop xds v2 support

### PR DESCRIPTION
**Pull request by YifeiZhuang**
https://github.com/grpc/proposal/pull/333

cc. @sergiitk

### CODE REVIEW

**Review by larry-safran — COMMENTED** [review]
Now that we will only have one version of LrsStream, it would simplify things if we eliminated the extra level represented by the abstract class.  It would be nice to move all of the logic in LrsStream into LrsStreamV3 and then renamed LrsStreamV3 to LrsStream.  Similarly on the tests, if we merged XdsClientImplTestBase and XdsClientImplV3Test it would make the logic clearer.

File: xds/src/test/java/io/grpc/xds/XdsClientImplV3Test.java [hunk 1]
```diff
@@ -177,6 +177,7 @@ public void cancelled(Context context) {
     };
   }
 
+
```
Comment (larry-safran) [inline xds/src/test/java/io/grpc/xds/XdsClientImplV3Test.java:180]: This extra blank line looks like an accident

**Review by larry-safran — COMMENTED** [review]
It looks to me that EnvoyProtoDataTest.java lines 90-123 should be deleted

File: xds/src/test/java/io/grpc/xds/XdsServerWrapperTest.java [hunk 2]
```diff
@@ -125,23 +125,42 @@ public void tearDown() {
   }
 
   @Test
+  @SuppressWarnings("unchecked")
   public void testBootstrap_notV3() throws Exception {
```
[the remark this answers is not in the document]
Comment (YifeiZhuang) [reply xds/src/test/java/io/grpc/xds/XdsServerWrapperTest.java:129]: Yea, renamed testBootstrap - it is a basic happy case of testing bootstrap.

**Comment by YifeiZhuang** [comment]
> Now that we will only have one version of LrsStream, it would simplify things if we eliminated the extra level represented by the abstract class. It would be nice to move all of the logic in LrsStream into LrsStreamV3 and then renamed LrsStreamV3 to LrsStream. Similarly on the tests, if we merged XdsClientImplTestBase and XdsClientImplV3Test it would make the logic clearer.

Thank you. Removing LrsStream abstract sounds fair. Will do in a separate PR.
For the XdsClientImplTestBase I am slightly inclined to keep it, for future usage. But if it causes too much trouble for us to maintain it, we will consider merging them.

**Review by larry-safran — APPROVED** [review]
LGTM

**Pull request merged**